### DISCUSSION TITLE
Отключение SSL Верификации

### DISCUSSION THREAD

**Discussion by Falconwatch** [ai-forever/gigachain#30] [Q&A] [answered]
Получаю ошибку ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1007)
Как отключить ssl верификацию?

from langchain.chat_models.gigachat import GigaChat
giga = GigaChat(profanity=False, credentials=...)

qa = ConversationalRetrievalChain.from_llm(giga, retriever=retriever)

questions = [
    "В каком году утонул титаник?",
    "Сколько шлюпок было на борту?",
    "Почему так много жертв?",
]
chat_history = []

for question in questions:
    result = qa({"question": question, "chat_history": chat_history})
    chat_history.append((question, result["answer"]))
    print(f"-> **Question**: {question} \n")
    print(f"**Answer**: {result['answer']} \n")

**Comment by Falconwatch** [comment, marked as answer]
Достаточно добавить в конструктор GigaChat параметр **verify_ssl_certs=False**
giga = GigaChat(profanity=False, 
                ssl_verify=False,
                **verify_ssl_certs=False,**
                credentials=...)

**Comment by AlexanderKozhevin** [comment]
@Falconwatch а можете подсказать где вы credentials брали. 
в смысле у меня уже есть акаунт и я дергал их API,  все ок. Но тот токен сюда не походит

**Reply by gritaro** [reply]
https://developers.sber.ru/docs/ru/gigachat/individuals-quickstart
здесь нужны **Авторизационные данные**
